File: libsignal/service/src/main/java/org/whispersystems/signalservice/internal/push/VerifyAccountResponse.java
package org.whispersystems.signalservice.internal.push;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

public class VerifyAccountResponse {
  @JsonProperty
  public String uuid;

  @JsonProperty
  public String pni;

  @JsonProperty
  public boolean storageCapable;

  @JsonProperty
  public String number;

  @JsonCreator
  public VerifyAccountResponse() {}

  public VerifyAccountResponse(String uuid, String pni, boolean storageCapable) {
    this.uuid           = uuid;
    this.pni            = pni;
    this.storageCapable = storageCapable;
  }

  public String getUuid() {
    return uuid;
  }

  public boolean isStorageCapable() {
    return storageCapable;
  }

  public String getPni() {
    return pni;
  }
}

File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/internal/push/VerifyAccountResponse.kt
/*
 * Copyright 2025 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.internal.push

class VerifyAccountResponse(
  val uuid: String? = null,
  val pni: String? = null,
  val storageCapable: Boolean = false,
  val number: String? = null,
  val reregistration: Boolean = false
)
